src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -1941,12 +1937,10 @@ public int setByte(DynamicObject string, int index, int value) {
             final Rope rope = rope(string);
             final int normalizedIndex = checkIndexNode.executeCheck(index, rope.byteLength());
 
-            final Rope left = leftSubstringNode.executeSubstring(rope, 0, normalizedIndex);
-            final Rope right = rightSubstringNode.executeSubstring(rope, normalizedIndex + 1, rope.byteLength() - normalizedIndex - 1);
-            final Rope middle = makeLeafRopeNode.executeMake(new byte[] { (byte) value }, rope.getEncoding(), CodeRange.CR_UNKNOWN, NotProvided.INSTANCE);
-            final Rope composed = composedConcatNode.executeConcat(middleConcatNode.executeConcat(left, middle, rope.getEncoding()), right, rope.getEncoding());
-
-            StringOperations.setRope(string, composed);
+            final Rope newRope = setByteNode.executeSetByte(rope, normalizedIndex, value);
+            if (newRope != rope) {
```

Comment:
Missing profile?

Reply:
Indeed. Although the compiler might be smart enough to not need it, but better be consistent.